src/screens/WhotoFollow/index.js

```diff
@@ -125,13 +145,27 @@ const WhotoFollow = () => {
     onLoad();
   }, []);
 
-  const handleSelected = (value) => {
+  const handleSelected = (value, section) => {
+    const sectionId = section?.id || section?.location_id || 'other';
+    const sectionIndex = getSectionIndex(sectionId);
+    const sectionData = getSectionData(sectionId);
+
     const copyFollowed = [...followed];
     const index = followed.indexOf(value);
     if (index > -1) {
       copyFollowed.splice(index, 1);
+      AnalyticsEventTracking.eventTrack(
+        BetterSocialEventTracking.ONBOARDING_WHO_TO_FOLLOW_USER_UNSELECTED
+      );
     } else {
       copyFollowed.push(value);
+      AnalyticsEventTracking.eventTrack(
+        BetterSocialEventTracking.ONBOARDING_WHO_TO_FOLLOW_USER_SELECTED,
+        {
+          categoryIndex: sectionIndex,
+          ...sectionData
+        }
+      );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `handleSelected` function now includes section data in its analytics tracking. This change enhances the functionality of the analytics tracking by providing more detailed information about the user's interactions. However, similar to the previous comment, the `users` array is being traversed multiple times within this function which could impact performance for large data sets. Consider optimizing this part of the code to reduce unnecessary iterations.

```diff
-  const handleSelected = (value, section) => {
-    const sectionId = section?.id || section?.location_id || 'other';
-    const sectionIndex = getSectionIndex(sectionId);
-    const sectionData = getSectionData(sectionId);
-
-    const copyFollowed = [...followed];
-    const index = followed.indexOf(value);
-    if (index > -1) {
-      copyFollowed.splice(index, 1);
-      AnalyticsEventTracking.eventTrack(
-        BetterSocialEventTracking.ONBOARDING_WHO_TO_FOLLOW_USER_UNSELECTED
-      );
-    } else {
-      copyFollowed.push(value);
-      AnalyticsEventTracking.eventTrack(
-        BetterSocialEventTracking.ONBOARDING_WHO_TO_FOLLOW_USER_SELECTED,
-        {
-          categoryIndex: sectionIndex,
-          ...sectionData
-        }
-      );
-    }
-    setFollowed(copyFollowed);
-  };
+  const handleSelected = (value, section) => {
+    const sectionId = section?.id || section?.location_id || 'other';
+    const { index: sectionIndex, ...sectionData } = getSectionData(sectionId);
+
+    const copyFollowed = [...followed];
+    const index = followed.indexOf(value);
+    if (index > -1) {
+      copyFollowed.splice(index, 1);
+      AnalyticsEventTracking.eventTrack(
+        BetterSocialEventTracking.ONBOARDING_WHO_TO_FOLLOW_USER_UNSELECTED
+      );
+    } else {
+      copyFollowed.push(value);
+      AnalyticsEventTracking.eventTrack(
+        BetterSocialEventTracking.ONBOARDING_WHO_TO_FOLLOW_USER_SELECTED,
+        {
+          categoryIndex: sectionIndex,
+          ...sectionData
+        }
+      );
+    }
+    setFollowed(copyFollowed);
+  };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->